core/chain_makers.go

```diff
@@ -196,14 +197,18 @@ func (b *BlockGen) OffsetTime(seconds int64) {
 // values. Inserting them into BlockChain requires use of FakePow or
 // a similar non-validating proof of work implementation.
 func GenerateChain(config *params.ChainConfig, parent *types.Block, engine consensus.Engine, db ethdb.Database, n int, gen func(int, *BlockGen)) ([]*types.Block, []types.Receipts) {
+	if config.ChainID == nil { // for testing purposes
```

Comment:
GenerateChain isn't being used, most likely it will be commented out or possibly completely deleted in a final refactor.

Reply:
> most likely it will be commented out or possibly completely deleted in a final refactor.

Is this PR not ready to go? I was under the impression the v1 generator is complete and ready to merge (pending this review). I agree this logic should be removed before merge